pkg/kubelet/config/file.go

```diff
@@ -62,26 +65,32 @@ func new(path string, nodeName types.NodeName, period time.Duration, updates cha
 		updates <- kubetypes.PodUpdate{Pods: pods, Op: kubetypes.SET, Source: kubetypes.FileSource}
 	}
 	store := cache.NewUndeltaStore(send, cache.MetaNamespaceKeyFunc)
-	return &sourceFile{
+	return &sourceFileListerWatcher{
 		path:           path,
 		nodeName:       nodeName,
+		period:         period,
 		store:          store,
 		fileKeyMapping: map[string]string{},
 		updates:        updates,
+		locker:         &sync.Mutex{},
 	}
 }
 
-func (s *sourceFile) run() {
-	if err := s.watch(); err != nil {
-		glog.Errorf("unable to read config path %q: %v", s.path, err)
-	}
+func (s *sourceFileListerWatcher) run() {
```

Comment:
It's confusing to have both run & watch. Can you ad some comments about why both are used?